casper/src/main/resources/PoS.rhox

```diff
@@ -344,17 +405,25 @@ new  PoS,
           }
         } |
 
-        contract deposit (@fromUserPk, @amount, returnCh) = {
+        contract deposit (@fromUserPk, @amount, @activeValidators, returnCh) = {
           new vaultCh,
               revAddressCh,
+              indexSender,
+              getBlockData(`rho:block:data`), blockDataCh,
               authKeyCh in {
             @RevVault!("deployerAuthKey", *authKeyCh) |
```

Comment:
I think this should be run concurrently with `revAddressOps!("fromPublicKey", ...)`. Not only does it increase concurrency, it also makes data dependencies syntactically evident. You should also review the Rholang you added to ensure that this is generally true. Same with the lexical scope comment below.

Reply:
This is still the case after your changes. I can't bear Rholang being harder to read than it already is.

Reply:
Sorry, I was sure that I had changed that already.